src/Component/CardStyle/CardStyle.tsx

```diff
@@ -0,0 +1,416 @@
+import RuleGenerator, { RuleGeneratorProps } from '../RuleGenerator/RuleGenerator';
+import { BulkEditor } from '../BulkEditor/BulkEditor';
+import IconSelector, { IconLibrary } from '../Symbolizer/IconSelector/IconSelector';
+import { ComparisonFilterProps } from '../Filter/ComparisonFilter/ComparisonFilter';
+
+// i18n
+export interface CardStyleLocale {
+  styleTitle: string;
+  classificationTitle: string;
+  multiEditTitle: string;
+  symbolizerTitle: string;
+  filterTitle: string;
+  iconLibrariesTitle: string;
+}
+
+// default props
+interface CardStyleDefaultProps {
+  /** The geoStylerStyle object */
+  style: GsStyle;
+  /** Locale object containing translated text snippets */
+  locale: CardStyleLocale;
+}
+
+// non default props
+export interface CardStyleProps extends Partial<CardStyleDefaultProps> {
+  /** Reference to internal data object (holding schema and example features) */
+  data?: Data;
+  /** The callback function that is triggered when the state changes */
+  onStyleChange?: (style: GsStyle) => void;
+  /** Properties of the filter components */
+  filterUiProps?: Partial<ComparisonFilterProps>;
+  /** The renderer to use for previews. */
+  rendererType?: 'SLD' | 'OpenLayers';
+  /** The passthrough props for the StyleOverview component. */
+  styleOverviewProps?: Partial<StyleOverviewProps>;
+  /** The passthrough props for the RuleOverview component. */
+  ruleOverviewProps?: Partial<RuleOverviewProps>;
+  /** The passthrough props for the RuleGenerator component. */
+  ruleGeneratorProps?: Partial<RuleGeneratorProps>;
+  /** The passthrough props for the Editor component. */
+  editorProps?: Partial<EditorProps>;
+  // /** List of supported icons ordered as library */
+  iconLibraries?: IconLibrary[];
+  /** Enable classification */
+  enableClassification?: boolean;
+}
+
+export interface CardView {
+  view: string;
+  props: any[];
+  path: Crumb[];
+}
+
+const STYLEVIEW = CardViewUtil.STYLEVIEW;
+const RULEVIEW = CardViewUtil.RULEVIEW;
+const CLASSIFICATIONVIEW = CardViewUtil.CLASSIFICATIONVIEW;
+const MULTIEDITVIEW = CardViewUtil.MULTIEDITVIEW;
+const SYMBOLIZERVIEW = CardViewUtil.SYMBOLIZERVIEW;
+const FILTEREDITVIEW = CardViewUtil.FILTEREDITVIEW;
+const ICONLIBRARIESVIEW = CardViewUtil.ICONLIBRARIESVIEW;
+
+export const CardStyle: React.FC<CardStyleProps> = ({
+  locale = en_US.GsCardStyle,
+  // enableClassification = true,
+  style = { name: 'My Style', rules: [] },
+  data,
+  onStyleChange,
+  filterUiProps,
+  rendererType,
+  iconLibraries,
+  enableClassification,
+  styleOverviewProps,
+  ruleOverviewProps,
+  editorProps,
+  ruleGeneratorProps
+}) => {
+
+  const defaultCrumb: Crumb = {view: STYLEVIEW, title: locale.styleTitle, indices: []};
+  const defaultView: CardView = {
+    view: STYLEVIEW,
+    props: [],
+    path: [defaultCrumb]
+  };
+  const [currentView, setCurrentView] = useState<CardView>(defaultView);
+
+  const getPathForView = (viewName: string, indices: number[]): Crumb[] => {
```

Comment:
I think this could be moved to the `CardViewUtil`.

Reply:
I'm planning on refactoring the view handling and providing a hook that manages this. This should cleanup the `CardStyle` component drastically